server/src/test/java/org/elasticsearch/snapshots/SnapshotResiliencyTests.java

```diff
@@ -1616,7 +1616,10 @@ private final class TestClusterNode {
             TestClusterNode(DiscoveryNode node) throws IOException {
                 this.node = node;
                 final Environment environment = createEnvironment(node.getName());
-                threadPool = deterministicTaskQueue.getThreadPool(runnable -> DeterministicTaskQueue.onNodeLog(node, runnable));
+                threadPool = deterministicTaskQueue.getThreadPool(
+                    runnable -> DeterministicTaskQueue.onNodeLog(node, runnable),
+                    System::nanoTime
+                );
```

Comment:
Am I correct that it is only new usage of `getThreadPool` with nano time supplier? Could you please clarify why that is required?

Reply:
You're correct. Please see full explanation in a the comment below